internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -659,3 +655,130 @@ func RestoreIrqBalanceConfig(ctx context.Context, irqBalanceConfigFile, irqBanne
 	}
 	return nil
 }
+
+// convertAnnotationToLatency converts the cpu-c-states.crio.io annotation to a maximum
+// latency value in microseconds.
+//
+// The cpu-c-states.crio.io annotation can be used to control c-states in several ways:
+//
+//	enable: enable all c-states (cpu-c-states.crio.io: "enable")
+//	disable: disable all c-states (cpu-c-states.crio.io: "disable")
+//	max_latency: enable c-states with a maximum latency in microseconds
+//	             (for example,  cpu-c-states.crio.io: "max_latency:10")
+//	cstate.id: enable c-states with a maximum kernel CPU idle level ID
+//	           (for example,  cpu-c-states.crio.io: "cstate.id:1")
+//	cstate.name: enable c-states using a case sensitive name of the maximum C-state allowed
+//	             (for example,  cpu-c-states.crio.io: "cstate.name:C1")
+//	by using 'None' as a fallback value to prevent errors when alternative C-state IDs/names do not exist.
+//	When 'None' is used in the alternatives pipeline, all the alternatives that follow 'None' are ignored.
+//
+// It is also possible to specify multiple fallback values separated by '|' as the C-state names and/or
+// IDs may not be available on some systems.
+//
+// Note: this is loosely based on how latencies are specified for the tuned cpu plugin.
+//
+// Examples:
+//
+// cpu-c-states.crio.io: "disable" (disable all c-states)
+// cpu-c-states.crio.io: "enable" (enable all c-states)
+// cpu-c-states.crio.io: "cstate.name:C6|cstate.id:3|max_latency:10" (Use the latency of C6 c-state. If C6 does not exist, use the latency of CPU idle level ID 3 (state3) in sysfs. If this is not found, use a latency of 10us.)
```

Comment:
I'm not 100% sure if GitHub/Go will be happy with ths, but here's a suggestion for consideration.

```suggestion
// cpu-c-states.crio.io: "cstate.name:C6|cstate.id:3|max_latency:10" (Use the latency of C6 c-state. If C6 does not exist, use the latency of CPU idle level ID 3 (state3) in sysfs. If this is not found, use a latency of 10μs.)
```

Reply:
Just for simplicity, I'd prefer not using special characters in the comments. Also "us" is already used for microseconds throughout the code in different identifiers and I'd like to be consistent.